File: app/src/main/java/com/example/gavin/gavinsource/MainActivity.java
package com.example.gavin.gavinsource;

import android.app.Fragment;
import android.app.FragmentTransaction;
import android.os.Bundle;
import android.view.View;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.alibaba.android.arouter.facade.annotation.Route;
import com.alibaba.android.arouter.launcher.ARouter;
import com.example.lib_link.constant.ActivityPath;
import com.example.lib_link.constant.DataKey;
import com.example.lib_link.constant.FragmentPath;
import com.example.common.base.BaseActivity;

import java.util.LinkedList;
import java.util.List;

import butterknife.BindView;
import butterknife.OnClick;


/**
 * @author gavin
 *         首页
 */
@Route(path = ActivityPath.MAIN_MAIN)
public class MainActivity extends BaseActivity {

    @BindView(R.id.fl_main)
    FrameLayout flMain;
    @BindView(R.id.iv_tab_one)
    ImageView ivTabOne;
    @BindView(R.id.tv_tab_one)
    TextView tvTabOne;
    @BindView(R.id.rv_tab_one)
    RelativeLayout rvTabOne;
    @BindView(R.id.iv_tab_center)
    ImageView ivTabCenter;
    @BindView(R.id.tv_tab_center)
    TextView tvTabCenter;
    @BindView(R.id.rv_tab_center)
    RelativeLayout rvTabCenter;
    @BindView(R.id.iv_tab_three)
    ImageView ivTabThree;
    @BindView(R.id.tv_tab_three)
    TextView tvTabThree;
    @BindView(R.id.rv_tab_three)
    RelativeLayout rvTabThree;
    @BindView(R.id.ll_tab)
    LinearLayout llTab;

    private List<Fragment> mainTabFragments = new LinkedList<>();

    @Override
    protected void onViewInjected() {

        initFragmentList();

        FragmentTransaction transaction = getFragmentManager().beginTransaction();
        transaction.replace(R.id.fl_main, mainTabFragments.get(0));
        transaction.addToBackStack(null);
        transaction.commit();

    }

    /**
     * 初始化首页可加载的Fragment列表
     */
    private void initFragmentList() {
        for (int i = 0; i < 3; i++) {
            Bundle bundle = new Bundle();
            bundle.putString(DataKey.KEY_TEST_ONE, "第" + i + "页");
            bundle.putString(DataKey.KEY_TEST_TWO, "第" + i + "页");

            String fragmentPath = i == 0 ? FragmentPath.MODULE_TAB_MAIN : FragmentPath.TEXT_ONE;
            Fragment tab = (Fragment) ARouter.getInstance()
                    .build(fragmentPath)
                    .navigation();
            if (tab != null) {
                tab.setArguments(bundle);
                mainTabFragments.add(tab);
            }
        }
    }

    @OnClick({R.id.rv_tab_one, R.id.rv_tab_center, R.id.rv_tab_three})
    public void onViewClicked(View view) {
        FragmentTransaction transaction = getFragmentManager().beginTransaction();
        if (view.getId() == R.id.rv_tab_one) {
            transaction.replace(R.id.fl_main, mainTabFragments.get(0));
            transaction.addToBackStack(null);
            transaction.commit();
        } else if (view.getId() == R.id.rv_tab_center) {
            transaction.replace(R.id.fl_main, mainTabFragments.get(1));
            transaction.addToBackStack(null);
            transaction.commit();
        } else if (view.getId() == R.id.rv_tab_three) {
            transaction.replace(R.id.fl_main, mainTabFragments.get(2));
            transaction.addToBackStack(null);
            transaction.commit();
        }
    }


    @Override
    protected int initContentViewId() {
        return R.layout.activity_main;
    }
}

File: app/src/main/java/com/gavin/whale/MainActivity.kt
package com.gavin.whale

import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import butterknife.OnClick
import com.alibaba.android.arouter.facade.annotation.Route
import com.alibaba.android.arouter.launcher.ARouter
import com.blankj.utilcode.util.BarUtils
import com.blankj.utilcode.util.FragmentUtils
import com.gavin.common.base.BaseActivity
import com.gavin.common.constants.ActivityPath
import com.gavin.common.constants.DataKey
import com.gavin.common.constants.FragmentPath
import java.util.*
import kotlin.collections.ArrayList
import kotlin.collections.HashMap


/**
 * @author com.gavin
 * 首页
 */
@Route(path = ActivityPath.APP_MAIN)
class MainActivity : BaseActivity() {

    private val mainTabFragments = LinkedHashMap<String,Fragment>()
    private var mShowFragmentPath = FragmentPath.TAB_I

    override fun onBaseInitAfter(savedInstanceState: Bundle?) {

        initFragments()
        FragmentUtils.add(supportFragmentManager, ArrayList(mainTabFragments.values), R.id.fl_main, 0)
        BarUtils.setStatusBarVisibility(this, false)
    }

    /**
     * 初始化首页可加载的Fragment列表
     */
    private fun initFragments() {
        for (i in 0..2) {
            val bundle = Bundle()
            bundle.putString(DataKey.KEY_TEST_ONE, "第" + i + "页")
            bundle.putString(DataKey.KEY_TEST_TWO, "第" + i + "页")
            var fragmentPath = ""
            when (i) {
                0 -> {
                    fragmentPath = FragmentPath.TAB_I
                }
                1 -> {
                    fragmentPath = FragmentPath.TAB_II
                }
                2 -> {
                    fragmentPath = FragmentPath.TAB_III
                }
            }

            val tab: Fragment? = ARouter.getInstance()
                    .build(fragmentPath)
                    .navigation() as Fragment?

            if (tab != null) {
                tab.arguments = bundle
                mainTabFragments[fragmentPath] = tab
            }
        }
    }

    @OnClick(R.id.rv_tab_one, R.id.rv_tab_center, R.id.rv_tab_three)
    fun onViewClicked(view: View) {
        var showFragmentPath = FragmentPath.PLACEHOLDER

        when (view.id) {
            R.id.rv_tab_one -> {
                showFragmentPath = FragmentPath.TAB_I
            }
            R.id.rv_tab_center -> {
                showFragmentPath = FragmentPath.TAB_II
            }
            R.id.rv_tab_three -> {
                showFragmentPath = FragmentPath.TAB_III
            }
        }
        var showFragment = ARouter.getInstance()
                .build(showFragmentPath)
                .navigation() as Fragment?
        if(showFragment == null){
            showFragment = ARouter.getInstance()
                    .build(FragmentPath.PLACEHOLDER)
                    .navigation() as Fragment
        }
        mShowFragmentPath = showFragmentPath
        FragmentUtils.showHide(showFragment, ArrayList(mainTabFragments.values))

    }


    override fun initContentViewId(): Int {
        return R.layout.activity_main
    }
}
